File: examples/insurance-claim-decision-agent-ts/src/prompts.ts
// Agent identity + model/prompt provenance. CLAUDE.md is the single source of
// truth for the operating instructions (shared verbatim with the Python example),
// so the canonical prompt_version_hash is sha256 over its raw bytes — a recorded
// decision pins exactly which prompt produced it, byte-identical across ports.
import { createHash } from "node:crypto";
import { readFileSync } from "node:fs";
import { fileURLToPath } from "node:url";
import { dirname, join } from "node:path";

const __dirname = dirname(fileURLToPath(import.meta.url));
const CLAUDE_MD = join(__dirname, "..", "CLAUDE.md");

const FALLBACK =
  "You are the Insurance Claim Decision Agent. Act only through the payer tools. " +
  "The model orchestrates and reasons; the deterministic engine computes the money; " +
  "a human owns every denial, reduction, or pend.";

let bytes: Buffer;
try {
  bytes = readFileSync(CLAUDE_MD);
} catch {
  bytes = Buffer.from(FALLBACK, "utf8");
}

export const AGENT_ID = "insurance-claim-decision-agent";
export const MODEL_VERSION = process.env.AGENT_MODEL || "claude-sonnet-4-6";
export const SYSTEM_PROMPT = bytes.toString("utf8");
export const PROMPT_VERSION_HASH = "sha256:" + createHash("sha256").update(bytes).digest("hex");

File: examples/insurance-claim-langgraph/agent/prompts.py
"""Agent identity + model/prompt provenance. CLAUDE.md is the operating manual
(shared verbatim with the other ports), so the canonical PROMPT_VERSION_HASH is a
sha256 over its bytes — byte-identical across every runtime."""
from __future__ import annotations

import hashlib
import os
from pathlib import Path

AGENT_ID = "insurance-claim-decision-agent"
MODEL_VERSION = os.environ.get("AGENT_MODEL", "claude-sonnet-4-6")
_CLAUDE_MD = Path(__file__).resolve().parent.parent / "CLAUDE.md"

try:
    _BYTES = _CLAUDE_MD.read_bytes()
except OSError:
    _BYTES = b"You are the Insurance Claim Decision Agent."

SYSTEM_PROMPT = _BYTES.decode("utf-8")
PROMPT_VERSION_HASH = "sha256:" + hashlib.sha256(_BYTES).hexdigest()
